torchtune/models/llama3/_tokenizer.py

```diff
@@ -226,28 +226,42 @@ def tokenize_message(
         Returns:
             List[int]: The list of token ids.
         """
-
         tokenized_header = self._tokenize_header(message) if tokenize_header else []
-
         tokenized_body = self._tokenize_body(message)
-
         tokenized_end = self._tokenize_end(message) if tokenize_end else []
 
         tokenized_message = tokenized_header + tokenized_body + tokenized_end
-
         return tokenized_message
 
     def tokenize_messages(
         self,
         messages: List[Message],
         add_eos: bool = True,
+        add_final_eot: bool = True,
     ) -> Tuple[List[int], List[bool]]:
         """
         Tokenize a list of messages into a list of token ids and masks.
 
         Args:
             messages (List[Message]): The list of messages to tokenize.
-            add_eos (bool): Wether to add the tokenizer's eos_id. Default True.
```

Comment:
Why not just reuse the tokenize_message "tokenize_end" flag and account for all ending tokens? You either want to continue from the middle of a message or you want a completed sequence.